sdk/azidentity/workload_identity.go

```diff
@@ -55,21 +59,31 @@ type WorkloadIdentityCredentialOptions struct {
 	// the application responsible for ensuring the configured authority is valid and trustworthy.
 	DisableInstanceDiscovery bool
 
-	// EnableAzureProxy determines whether the credential reads proxy configuration from environment variables. When
-	// this value is true and proxy configuration isn't present or this value is false, the credential will request
+	// EnableAzureProxy determines whether the credential reads proxy configuration from environment variables or
+	// from the AzureProxy field.
+	// When this value is true and proxy configuration isn't present or this value is false, the credential will request
 	// tokens directly from Entra ID.
 	//
 	// The proxy feature is designed for applications that deploy to many clusters and clusters that host many
 	// applications. See the Azure Kubernetes Service identity bindings documentation for more information on when
 	// to set this option: https://learn.microsoft.com/azure/aks/identity-bindings-concepts
 	EnableAzureProxy bool
 
+	// AzureProxy specifies the options for the Azure proxy.
+	// If EnableAzureProxy is false, this field is ignored.
+	AzureProxy WorkloadIdentityAzureProxyOptions
```

Comment:
```suggestion
	AzureProxy customtokenproxy.Options
```

And delete the `WorkloadIdentityAzureProxyOptions` alias.